src/Features/Core/Portable/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs

```diff
@@ -150,6 +161,28 @@ protected bool TrySimplifyTypeNameExpression(SemanticModel model, SyntaxNode nod
             builder["OptionName"] = nameof(CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInMemberAccess); // TODO: need the actual one
             builder["OptionLanguage"] = model.Language;
             diagnostic = DiagnosticHelper.Create(descriptor, tree.GetLocation(issueSpan), severity, additionalLocations: null, builder.ToImmutable());
+
+#if LOG
+            var logLine = tree.FilePath + "\t" + diagnosticId + "\t" + inDeclaration + "\t";
+            var sourceText = tree.GetText(cancellationToken);
+            sourceText.GetLineAndOffset(issueSpan.Start, out var startLineNumber, out var startOffset);
+            sourceText.GetLineAndOffset(issueSpan.End, out var endLineNumber, out var endOffset);
+
+            var leading = sourceText.ToString(TextSpan.FromBounds(
+                sourceText.Lines[startLineNumber].Start, issueSpan.Start));
+            var mid = sourceText.ToString(issueSpan);
+            var trailing = sourceText.ToString(TextSpan.FromBounds(
+                issueSpan.End, sourceText.Lines[endLineNumber].End));
+
+            var contents = leading + "[|" + s_newlinePattern.Replace(mid, " ") + "|]" + trailing;
+            logLine += contents + "\r\n";
+
+            lock (_gate)
+            {
+                File.AppendAllText(_logFile, logLine);
+            }
+#endif
```

Comment:
this log has been useful for running additional analysis passes on SimplifyTypeNames and finding several bugs. I would like to leave it in (disabled) so i don't have to constantly add/remove the code.

Reply:
Is it possible to derive this log information from the diagnostics themselves? If so, we could make a new mode of AnalyzerRunner to handle this.

Reply:
Some of it can be cleaned from the diagnostics.  But not all of what i dump here.  Some of the data i'm collecting here is internal to this feature and is 'lost' by the time we have diagnostics.  Technically i could maybe store it int he `properties` of hte diagnostics.  However, i'm a bit worried about that as it might impact things in unexpected ways (more allocs/overhead/etc.)

Reply:
Note: i am open to the idea of exploring how we could make a generalized systme here.  but i also don't want to slow down this current effort too much :)